app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -40,11 +49,50 @@ public int getLevelsCount() {
         return levelsCount;
     }
 
-    public int getCurrentLevel() {
-        return currentLevel;
+    public int getCurrentLevelFromSharedPreferences() {
+        return gameData.getInt(GAME_VIEW_CURRENT_LEVEL_STRING, /* defValue= */ 1);
+    }
+
+    public int getLastLevelFromSharedPreferences() {
+        return gameData.getInt(GAME_VIEW_LAST_LEVEL_STRING, /* defValue= */ getCurrentLevel());
+    }
+
+    public void increaseLevel() {
+        if (currentLevel < levelsCount) {
+            SharedPreferences.Editor editor = gameData.edit();
+            // Increasing the current level
+            editor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, ++currentLevel);
+            // Increasing the last level
+            if (currentLevel > lastLevel) {
+                lastLevel = currentLevel;
+                editor.putInt(GAME_VIEW_LAST_LEVEL_STRING, currentLevel);
+            }
+            editor.apply();
+        }
     }
 
     public void setCurrentLevel(int currentLevel) {
         this.currentLevel = currentLevel;
     }
+
+    public void setLastLevel(int lastLevel) {
+        this.lastLevel = lastLevel;
+    }
+
+    public void resetGameData() {
+        currentLevel = 1;
+        lastLevel = 1;
+        SharedPreferences.Editor editor = gameData.edit();
+        editor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, 1);
```

Comment:
nit: put currentLevel and lastLevel instead. It will help if you decide that the first default level will be different.

Reply:
Thanks, fixed.